app/uk/gov/hmrc/gform/models/FormModel.scala

```diff
@@ -234,6 +236,14 @@ case class FormModel[A <: PageMode](
         case head :: _ => availableSectionNumbers.reverse.find(_ < sectionNumber).getOrElse(head)
       }
     }
+  def maybeSectionNumbersFrom(fcId: FormComponentId): List[SectionNumber] =
+    pageLookup
+      .get(fcId)
+      .map { pageModel =>
+        pagesWithIndex.collect { case (pm, sn) if pm == pageModel => sn }
```

Comment:
If we need to get a `SectionNumber` for given `FormComponentId` we can write dedicated lookup for it, rather than doing `pm == pageModel` check.